src/Http/Controllers/Admin/FeaturedController.php

```diff
@@ -128,7 +187,7 @@ private function getFeaturedSources($featuredSection, $search = null)
                     $bucketable['with'] ?? [],
                     ($bucketable['scopes'] ?? []) + ($scopes ?? []),
                     $bucketable['orders'] ?? [],
-                    $bucketable['per_page'] ?? request('offset') ?? 10,
+                    $bucketable['per_page'] ?? $request->get('offset') ?? 10,
```

Comment:
Making a parameter named `offset` just like the `search` parameter is more logical. `getFeaturedSources` doesn't need to know about the `request`. Also replace the `$request->get('offset') ?? 10` section with just `$offset` and set `$offset` parameter's default value as `10` in the function definition.

This is a good reason for using DI instead of helpers as it shows this kind of code smells.

Reply:
You are right and I am always voting up this kind of proposals, but let's not extend the scope of this PR, can you move it to another PR?

Reply:
I think we need a general refactoring issue/PR and put general changes into there and see if there are other minimal refactors that we can do. Your ones are specific about the things that aimed to be refactored.

Reply:
Well, I rather think that we need feature-specific refactoring related issues and not general ones, since we can loose the track of what's changed and what's needed to be changed. We need to document the standard of how things should be done and that's what the issues are for. As you can see, I've documented this one before starting working on it and I made a plan, which have been approved before I started working on it. You can do the same. What I wanted to point out is that you should first propose a change in a separate issue and let others tell their opinions, especially people from the core team. I don't want to offend you but you requested a lot of changes here which were inappropriate and inaccurate. I see you have a decent knowledge of Laravel but I think you haven't gone that far with Twill, so let's don't make a mess of this one which already has over 1.5k add's and 0.5k del's. Hope you can understand. 😄

Reply:
I know its so much better to categorize things but splitting or grouping the parameters, reducing some if statements or extracting method, etc. is fitting the `general` category for me. Any advice on categorizing things is welcome.

> I don't want to offend you but you requested a lot of changes here which were inappropriate and inaccurate. I see you have a decent knowledge of Laravel but I think you haven't gone that far with Twill, so let's don't make a mess of this one which already has over 1.5k add's and 0.5k del's. Hope you can understand. 😄

You are right. I am sorry, I haven't done anything about Twill so far, some concepts and the UI/UX stole my heart but still tried to review a PR about Twill.
I detected that some things that can be improved about CLI, documentation, code base. Your active contributions and the things that I like about Twill fueled my enthusiasm. My enthusiasm about **clean code** and such is high, I can't hold my self.
I feel bad because I wasted some of your time. It's not just Twill, I am not much experienced with Laravel either. I started learning it to make a project last year and created a project or two until today. I am just a university student that finished the 4th semester, didn't get any course in university about web development but love these things. I will try not to make contributions other than opening some issues, maybe simple PR's and giving emoji reactions to comments.

Reply:
@yusufkandemir Your enthusiasm and help is always appreciated. I completely understand you. You said that you created only two projects in Laravel, compared to what you said here in the PR, I can see you are doing a great job and you can become pretty well proficient very soon. I really respect every opinion and your opinion here was still very helpful. Don't worry and thank you!

Twill doesn't have a huge codebase but it was designed to handle a lot completely different cases and situations so everything written here is very tightly coupled (_product oriented_) and it has to be like that. That's one of the valid approaches for designing and building a product like this one. Another valid approach would be to decouple and abstract everything in the code base into much smaller groups of functionalities and that would be something we could call `framework oriented`. That's the pure essence of a very complex procedure called _refactoring_. Refactoring requires a lot of time, patience, knowledge and experience. It's a whole another type of work, it's even not that much software programming, but rather software architecture/design type of work. So, as I could see you are very up to contribute here in the terms of cleaning the code, restructuring, etc. That's excellent and you should keep up with it. But we need to do that step-by-step, patiently and with acceptance from other members/contributors. Since this is not just yours and mine project, we have to let others know what exactly we are proposing, why are we doing that and what is that going to affect. I am aware that there are a lot of possibilities for improvements but I am not proposing everything at once by purpose - so others can spontaneously and _naturally_ get involved and not get confused. I suggest you to do so.

I am always available for help and if you have any question, feel free to contact me here or on Spectrum.
Thank you and good luck!